bigchaindb/util.py

```diff
@@ -22,3 +30,191 @@ def start(self):
+    """Create a new transaction
+
+    A transaction in the bigchain is a transfer of a digital asset between two entities represented
+    by public keys.
+
+    Currently the bigchain supports two types of operations:
+
+        `CREATE` - Only federation nodes are allowed to use this operation. In a create operation
+        a federation node creates a digital asset in the bigchain and assigns that asset to a public
+        key. The owner of the private key can then decided to transfer this digital asset by using the
+        `transaction id` of the transaction as an input in a `TRANSFER` transaction.
+
+        `TRANSFER` - A transfer operation allows for a transfer of the digital assets between entities.
+
+    Args:
+        current_owner (str): base58 encoded public key of the current owner of the asset.
+        new_owner (str): base58 encoded public key of the new owner of the digital asset.
+        tx_input (str): id of the transaction to use as input.
+        operation (str): Either `CREATE` or `TRANSFER` operation.
+        payload (Optional[dict]): dictionary with information about asset.
+
+    Returns:
+        dict: unsigned transaction.
+
+
+    Raises:
+        TypeError: if the optional ``payload`` argument is not a ``dict``.
+    """
+
+    data = None
+    if payload is not None:
+        if isinstance(payload, dict):
+            hash_payload = hash_data(serialize(payload))
+            data = {
+                'hash': hash_payload,
+                'payload': payload
+            }
+        else:
+            raise TypeError('`payload` must be an dict instance')
+
+    hash_payload = hash_data(serialize(payload))
+    data = {
+        'hash': hash_payload,
+        'payload': payload
+    }
+
+    tx = {
+        'current_owner': current_owner,
+        'new_owner': new_owner,
+        'input': tx_input,
+        'operation': operation,
+        'timestamp': timestamp(),
+        'data': data
+    }
+
+    # serialize and convert to bytes
+    tx_serialized = serialize(tx)
+    tx_hash = hash_data(tx_serialized)
+
+    # create the transaction
+    transaction = {
+        'id': tx_hash,
+        'transaction': tx
+    }
+
+    return transaction
+
+
+def sign_tx(transaction, private_key):
+    """Sign a transaction
+
+    A transaction signed with the `current_owner` corresponding private key.
+
+    Args:
+        transaction (dict): transaction to sign.
+        private_key (str): base58 encoded private key to create a signature of the transaction.
+
+    Returns:
+        dict: transaction with the `signature` field included.
+
+    """
+    private_key = PrivateKey(private_key)
+    signature = private_key.sign(serialize(transaction))
+    signed_transaction = transaction.copy()
+    signed_transaction.update({'signature': signature})
+    return signed_transaction
+
+
+def create_and_sign_tx(private_key, current_owner, new_owner, tx_input, operation='TRANSFER', payload=None):
```

Comment:
maybe misleading, as it is a `operation=TRANSFER` by default and does require a tx_input

maybe remove the `operation` from the kwargs?